heat_infoblox/resources/bgp.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) 2016 Infoblox Inc.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import logging
+
+from heat.common.i18n import _
+from heat.engine import constraints
+from heat.engine import properties
+from heat.engine import resource
+from heat.engine import support
+from oslo_concurrency import lockutils
+
+from heat_infoblox import constants
+from heat_infoblox import resource_utils
+
+LOG = logging.getLogger(__name__)
+
+
+class Bgp(resource.Resource):
+    """A resource which represents a BGP Neighbor configuration.
+
+    This resource is used configure a BGP Autonomous System and single BGP
+    Neighbor per resource.
```

Comment:
Yeah, we need to split this into two resources. That's what I was expecting to see - sorry if there was a misunderstanding or I wasn't clear.

There is no reason you can't do this all in one stack still, but if you have separate neighbors you can easily add and remove neighbors too. This is how I did the NS group and NS group members. I would expect something similar here.

When creating the neighbor, you have an AS property. If that AS doesn't exist on the grid already, then it's an error. In your templates, you'll need to use the depends_on property if the template creates both the AS and the neighbor entries themselves.